core/finality-grandpa/src/environment.rs

```diff
@@ -429,7 +433,7 @@ where
 		debug!(target: "afg", "Finding best chain containing block {:?} with number limit {:?}", block, limit);
 
 		match self.select_chain.finality_target(block, None) {
-			Ok(Some(mut best_hash)) => {
+			Ok(Some(best_hash)) => {
```

Comment:
I guess in this context I don't understand the purpose of `finality_target` anymore.

Reply:
I agree, we already had a discussion about `SelectChain` on https://github.com/paritytech/substrate/pull/2869#issuecomment-502267671. Since we now have voting rules to implement the vote target logic what do you think about renaming `finality_target` in `SelectChain` to `best_containing`? In this case `SelectChain` is only responsible for implementing some custom notion of "best".